File: bindings/lua/Polycode/SkeletonAnimation.lua
class "SkeletonAnimation"


function SkeletonAnimation:SkeletonAnimation(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SkeletonAnimation(unpack(arg))
	end
end

function SkeletonAnimation:addBoneTrack(boneTrack)
	local retVal = Polycode.SkeletonAnimation_addBoneTrack(self.__ptr, boneTrack.__ptr)
end

function SkeletonAnimation:getName()
	local retVal =  Polycode.SkeletonAnimation_getName(self.__ptr)
	return retVal
end

function SkeletonAnimation:Play(once)
	local retVal = Polycode.SkeletonAnimation_Play(self.__ptr, once)
end

function SkeletonAnimation:Stop()
	local retVal =  Polycode.SkeletonAnimation_Stop(self.__ptr)
end

function SkeletonAnimation:Reset()
	local retVal =  Polycode.SkeletonAnimation_Reset(self.__ptr)
end

function SkeletonAnimation:Update()
	local retVal =  Polycode.SkeletonAnimation_Update(self.__ptr)
end

function SkeletonAnimation:setSpeed(speed)
	local retVal = Polycode.SkeletonAnimation_setSpeed(self.__ptr, speed)
end

function SkeletonAnimation:setWeight(newWeight)
	local retVal = Polycode.SkeletonAnimation_setWeight(self.__ptr, newWeight)
end

function SkeletonAnimation:getWeight()
	local retVal =  Polycode.SkeletonAnimation_getWeight(self.__ptr)
	return retVal
end

function SkeletonAnimation:isPlaying()
	local retVal =  Polycode.SkeletonAnimation_isPlaying(self.__ptr)
	return retVal
end

function SkeletonAnimation:__delete()
	if self then Polycode.delete_SkeletonAnimation(self.__ptr) end
end

File: bindings/javascript/Polycode/SkeletonAnimation.js
function SkeletonAnimation(name,duration) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SkeletonAnimation(name,duration)
	}
}


Duktape.fin(SkeletonAnimation.prototype, function (x) {
	if (x === SkeletonAnimation.prototype) {
		return;
	}
	Polycode.SkeletonAnimation__delete(x.__ptr)
})

SkeletonAnimation.prototype.addBoneTrack = function(boneTrack) {
	Polycode.SkeletonAnimation_addBoneTrack(this.__ptr, boneTrack.__ptr)
}

SkeletonAnimation.prototype.getName = function() {
	return Polycode.SkeletonAnimation_getName(this.__ptr)
}

SkeletonAnimation.prototype.Play = function(once) {
	Polycode.SkeletonAnimation_Play(this.__ptr, once)
}

SkeletonAnimation.prototype.Stop = function() {
	Polycode.SkeletonAnimation_Stop(this.__ptr)
}

SkeletonAnimation.prototype.Reset = function() {
	Polycode.SkeletonAnimation_Reset(this.__ptr)
}

SkeletonAnimation.prototype.Update = function() {
	Polycode.SkeletonAnimation_Update(this.__ptr)
}

SkeletonAnimation.prototype.setSpeed = function(speed) {
	Polycode.SkeletonAnimation_setSpeed(this.__ptr, speed)
}

SkeletonAnimation.prototype.setWeight = function(newWeight) {
	Polycode.SkeletonAnimation_setWeight(this.__ptr, newWeight)
}

SkeletonAnimation.prototype.getWeight = function() {
	return Polycode.SkeletonAnimation_getWeight(this.__ptr)
}

SkeletonAnimation.prototype.isPlaying = function() {
	return Polycode.SkeletonAnimation_isPlaying(this.__ptr)
}
